megamek/src/megamek/common/Tank.java

```diff
@@ -958,7 +960,7 @@ public HitData rollHitLocation(int table, int side, int aimedLocation,
                 // the opposite direction to which it entered the hex it went
                 // Hull Down in hit side they come in from
                 else {
-                    nArmorLoc = side;
+                    nArmorLoc = SIDE_LOC_MAPPING[side];
```

Comment:
What about superheavy tanks/large support vehicles, which have two additional side locations?

Reply:
Good question, I'll have to do some research on that one.

Reply:
As it turns out, superheavy tanks and large support vehicles can't go "hull down", but are effectively subject to partial cover rules (TacOps 2018 reprint). That's a little beyond the scope of this change, but I did introduce a mapping in the relevant class for future use.